src/database/schema/ChatSchema.ts

```diff
@@ -101,9 +101,8 @@ class ChatSchema implements BaseDbSchema {
 
   saveIfNotExist = async (db: SQLiteDatabase) => {
     try {
-      const existingChat = await ChatSchema.getByid(db, this.id);
+      const existingChat = await ChatSchema.isExists(db, this.id);
       if (!existingChat) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `isExists` method is used to check if a chat exists in the database. However, it seems like there's no error handling for this operation. If the `isExists` method fails due to any reason (like a database connection issue), it might lead to unexpected behavior. Consider adding error handling here.
```diff
  try {
+    const existingChat = await ChatSchema.isExists(db, this.id);
+    if (!existingChat) {
+      await this.save(db);
+    }
+  } catch (e) {
+    // Handle or throw the error
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->